src/main/java/com/aws/greengrass/clientdevices/auth/iot/GreengrassV2DataClientFactory.java

```diff
@@ -53,8 +53,9 @@ public GreengrassV2DataClient getClient() throws DeviceConfigurationException {
 
         GreengrassV2DataClientBuilder clientBuilder =
                 GreengrassV2DataClient.builder().credentialsProvider(AnonymousCredentialsProvider.create())
-                        .httpClient(httpClient.build()).overrideConfiguration(
-                                ClientOverrideConfiguration.builder().retryPolicy(RetryPolicy.none()).build());
+                        .httpClientBuilder(httpClient.useIdleConnectionReaper(false))
```

Comment:
Why is this set to false?

Reply:
Explained in description. We aren't pooling our clients, so there's no reason why we should use the idle reaper. Setting this to false will also avoid the leak (in addition to using `httpClientBuilder`)

Reply:
I believe that forces the http connection to be closed when the `GreengrassV2DataClient` get closed instead of waiting for it to be closed async which I believe is what we want given if this gets called too frequently we might end up with too many open http resources